src/Microsoft.AspNet.OData.Shared/Formatter/EdmLibHelpers.cs

```diff
@@ -756,6 +756,12 @@ public static bool HasPrecision(EdmPrimitiveTypeKind primitiveTypeKind)
                     primitiveTypeKind == EdmPrimitiveTypeKind.TimeOfDay);
         }
 
+        public static IEdmTypeReference GetTypeReferenceOfProperty(IEdmModel model, Type clrType, string propertyName)
+        {
+            PropertyInfo propertyInfo = clrType.GetProperty(propertyName);
```

Comment:
>clrType.GetProperty(propertyName); [](start = 40, length = 34)

where did you get the "propertyName"? 

be caution that the property from Edm model maybe different with the property name on the CLR type.

Reply:
We use the CLR property name. It is coming from the generated skiptoken value. 

---
In reply to: [281398993](https://github.com/OData/WebApi/pull/1724#discussion_r281398993) [](ancestors = 281398993)